packages/react-components/src/listbox/src/Listbox.tsx

```diff
@@ -33,81 +36,96 @@ const SelectionMode = {
+    /**
+     * Called when the focus change.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {String[]} keys - The keys of the selected items.
+     * @returns {void}
+     */
+    onFocusChange?(event: SyntheticEvent, key: string, activeElement: HTMLElement): void;
     /**
      * A controlled set of the selected item keys.
      */
-    selectedKeys: arrayOf(string),
+    selectedKeys?: string[]
     /**
      * The initial value of `selectedKeys` when uncontrolled.
      */
-    defaultSelectedKeys: arrayOf(string),
+    defaultSelectedKeys?: string[]
     /**
      * Called when the selected keys change.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @param {String[]} keys - The keys of the selected items.
      * @returns {void}
      */
-    onSelectionChange: func,
+    onSelectionChange?(event: SyntheticEvent, key: string[]): void;
     /**
      * The type of selection that is allowed.
      */
-    selectionMode: oneOf(["none", "single", "multiple"]),
+    selectionMode?: "none" | "single" | "multiple";
     /**
      * A collection of nodes to render instead of children. It should only be used if you embed a Listbox inside another component like a custom Select.
      */
-    nodes: arrayOf(object),
+    nodes?: CollectionNode[]
     /**
      * Whether or not the listbox should autofocus on render.
      */
-    autoFocus: oneOfType([bool, number]),
+    autoFocus?: boolean | number;
     /**
      * Default focus target when enabling autofocus.
      */
-    defaultFocusTarget: string,
+    defaultFocusTarget?: string;
     /**
      * Whether or not to focus the hovered item.
      */
-    focusOnHover: bool,
+    focusOnHover?: boolean;
     /**
      * Whether or not focus should be virtual (add a CSS class instead of switching the active element).
      */
-    useVirtualFocus: bool,
+    useVirtualFocus?: boolean;
     /**
      * Whether or not the listbox option should be reachable with tabs.
      */
-    tabbable: bool,
+    tabbable?: boolean;
     /**
      * Whether or not the listbox take up the width of its container.
      */
-    fluid: bool,
+    fluid?: boolean;
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    as: oneOfType([string, elementType]),
+    as?: ElementType;
     /**
-     * React children.
+     * @ignore
      */
-    children: oneOfType([any, func])
-};
+    children?: ReactNode;
+    /**
+     * @ignore
+     */
+    forwardedRef: ForwardedRef<any>
+}
 
-function useCollectionNodes(children, nodes) {
+function useCollectionNodes(children: ReactNode, nodes: CollectionNode[]) {
     const collectionNodes = useCollection(children);
 
     return nodes ?? collectionNodes;
 }
 
-function useSelectionManager(items, { selectedKeys }) {
+function useSelectionManager(items: CollectionNode[], { selectedKeys }: { selectedKeys?: string[] }) {
```

Comment:
At this point, non items node has been filtered out, so it's only CollectionItem :)

Reply:
done